doc/source/overview/high_level.md

```diff
@@ -0,0 +1,508 @@
+
+$$
+f(S) = \mathbb{E}_{D}[f(x)|x_S]
+$$
+
+Computing this value is very difficult. Instead, we can approximate the above using the interventional conditional expectation, which is defined as:
+
+$$
+f(S) = \mathbb{E}_{D}[f(x)|do(x_S)]
+$$
+
+The $do$ operator here fixes the values of the features in $S$ and samples the remaining $\bar{S}$ feature values from the data. A Downside of interfering in the distribution like this can mean introducing unrealistic samples if there are dependencies between the features.
+
+**Pros**
+- The Shapley values are fairly distributed among the feature values
+- Shapley values can be easily interpreted and visualized
+- Very general as is a blackbox method
+
+**Cons**
+- KernalSHAP is slow owing to the number of samples required to estimate the Shapley values accurately
+- The interventional conditional probability introduces unrealistic data points
+- Requires access to the training dataset
+
+
+#### TreeSHAP
+
+| Model-types  | Task-types     | Data-types  |
+| ------------ | -------------- | ----------- |
+| Tree-based   | Classification | Tabular     |
+|              | Regression     | Categorical |
+
+
+In the case of tree-based models, we can obtain a speed-up by exploiting the structure of trees. Alibi exposes two white-box methods, Interventional and Path dependent feature perturbation. The main difference is that the path-dependent method approximates the interventional conditional expectation, whereas the interventional method calculates it directly.
+
+#### Path Dependent TreeSHAP
+
+Given a coalition, we want to approximate the interventional conditional expectation. We apply the tree to the features present in the coalition like we usually would, with the only difference being when a feature is missing from the coalition. In this case, we take both routes down the tree, weighting each by the proportion of samples from the training dataset that go each way. For this algorithm to work, we need the tree to record how it splits the training dataset. We don't need the dataset itself, however, unlike the interventional TreeSHAP algorithm.
+
+Doing this for each possible set $S$ involves $O(TL2^M)$ time complexity. We can significantly improve the algorithm to polynomial-time by computing the path of all sets simultaneously. The intuition here is to imagine standing at the first node and counting the number of subsets that will go one way, the number that will go the other, and the number that will go both (in the case of missing features). Because we assign different sized subsets different weights, we also need to distinguish the above numbers passing into each tree branch by their size. Finally, we also need to keep track of the proportion of sets of each size in each branch that contains a feature $i$ and the proportion that don't. Once all these sets have flowed down to the leaves of the tree, then we can compute the Shapley values. Doing this gives us $O(TLD^2)$ time complexity.
```

Comment:
Probably no need to provide this intuition on this doc, citing the big-oh time complexity decrease from naive implementation to polynomial time should be enough with a link to the docs.